src/java/org/apache/cassandra/schema/Schema.java

```diff
@@ -606,11 +606,16 @@ public synchronized void clear()
      * in-memory representation got out of sync somehow with what's on disk.
      */
     public synchronized void reloadSchemaAndAnnounceVersion()
+    {
+        reloadSchema();
+        updateVersionAndAnnounce();
+    }
+
+    public synchronized void reloadSchema()
     {
         Keyspaces before = keyspaces.filter(k -> !SchemaConstants.isLocalSystemKeyspace(k.name));
         Keyspaces after = SchemaKeyspace.fetchNonSystemKeyspaces();
         merge(Keyspaces.diff(before, after));
-        updateVersionAndAnnounce();
```

Comment:
I'm not sure about that, but I was wondering whether we should update the version after replaying the commit log? I justify my question by the fact that schema and its version is read before replaying the commit log, if there are mutations replayed, the versions is likely to change - by recalculating it, we may avoid unnecessary syncing schema with other nodes - does it make sense?

Reply:
I think that the version is updated and announced by the call to `StorageService.initServer -> StorageService.prepareToJoin -> Schema.updateVersionAndAnnounce` that is done by `CassandraDaeamon#setup` [here](https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/service/CassandraDaemon.java#L417), after the commitlog has been replayed some lines above, [here](https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/service/CassandraDaemon.java#L357). So if I'm understanding it correctly the version is updated after replaying the commit log.